src/plugins/intel_gpu/src/kernel_selector/cl_kernels/convolution_gpu_bfyx_os_iyx_osv16.cl

```diff
@@ -113,8 +115,8 @@ KERNEL(convolution_gpu_bfyx_os_iyx_osv16)(
         for(uint in_block_pos = 0; in_block_pos < IN_BLOCK_ARRAY_SIZE * SUB_GROUP_SIZE; in_block_pos += SUB_GROUP_SIZE) {
             // Horizontal position in input block after read.
             const uint in_block_next_x_pos = in_block_pos % IN_BLOCK_WIDTH + SUB_GROUP_SIZE;
-
-            in[in_block_pos / SUB_GROUP_SIZE] = input[tmp_in_addr + (in_block_pos % IN_BLOCK_WIDTH) * INPUT0_X_PITCH];
+            const int idx = tmp_in_addr + (in_block_pos % IN_BLOCK_WIDTH) * INPUT0_X_PITCH;
+            in[in_block_pos / SUB_GROUP_SIZE] = idx < input0_physical_len ? input[idx] : UNIT_VAL_ZERO;
```

Comment:
Does it impact the performance? Maybe we need to have separate compile-time branch which handle remainder?

Reply:
updated the code to remove conditional statement and used `min` instead. no perf issue from fp16 static model performance test on iGPU.